hooks/update_gh_pages.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script may be used as pre commit hook
+
+import json
+import subprocess
+from typing import Sequence, Set, Dict
+
+
+IGNORED_DIRS = {'docs', 'hooks'}
+
+
+def get_repo_path() -> str:
+    return subprocess.check_output(['git', 'rev-parse', '--show-toplevel'], encoding='utf-8').strip()
+
+
+def get_repository_filenames(repo_path: str):
+    return subprocess.check_output(['git', 'ls-files'], encoding='utf-8', cwd=repo_path).split('\n')
+
+
+def get_template_dirs(repository_filenames: Sequence[str]) -> Set[str]:
+    template_dirs = set()
+    for filename in repository_filenames:
+        dir_name = filename.split('/')[0]
+        if '/' in filename and dir_name not in IGNORED_DIRS:
+            template_dirs.add(f'{dir_name}')
+
+    return template_dirs
+
+
+def pack_template_dirs(repo_path: str, template_dirs: Set[str]):
+    for template_dir in template_dirs:
+        subprocess.check_call(['tar', '-cvjf', f'{repo_path}/docs/{template_dir}.tar.bz2', '-C',
+                               f'{repo_path}', f'{template_dir}'])
+
+
+def parse_template_metadata(chart_yaml_path: str, metadata_fields: Set[str]) -> Dict[str, str]:
+    template_metadata = {}
+    with open(chart_yaml_path, mode='r', encoding='utf-8') as chart_file:
+        for line in chart_file.readlines():
```

Comment:
Maybe you should consider using yaml.safe_load(chart_file) - it seems to be more convenient.